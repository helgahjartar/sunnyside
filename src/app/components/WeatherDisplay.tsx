'use client';

import { useWeather } from '../contexts/WeatherContext';
import { ForecastType } from '../utils/types';
import WeatherDetails from './WeatherDetails';

const ForecastDisplay = () => {
   const { todaysWeather, forecastType, setForecastType, weatherForecast } = useWeather();

   // Handler for forecast type toggling
   const toggleForecastType = () => {
      setForecastType(
         forecastType === ForecastType.OneDay ? ForecastType.FiveDay : ForecastType.OneDay
      );
   };

   // Simple loading component
   const Loading = (
      <>
         <div className="text-2xl font-semibold text-black">Loading..</div>
      </>
   );

   return (
      <div>
         <div className="flex flex-row justify-between items-center">
            <div className="h-12 text-black text-3xl font-bold w-full truncate py-2 px-4 capitalize">
               {todaysWeather && (
                  <>
                     {todaysWeather?.city}, {todaysWeather?.country}
                  </>
               )}
            </div>
            <div className="flex align-middle">
               <button
                  className="min-h-10 bg-orange-300 hover:bg-orange-500 text-white text-lg font-bold py-2 px-4 rounded-lg shadow-lg transition duration-300 ease-in-out transform hover:scale-105"
                  onClick={toggleForecastType}
               >
                  {forecastType === ForecastType.OneDay
                     ? 'See 5-day Forecast'
                     : 'See Current Weather'}
               </button>
            </div>
         </div>
         <div
            className={`flex justify-between overflow-scroll mt-4 bg-[#F5ECDA] rounded-lg py-8 px-10 space-x-10 ${
               forecastType === ForecastType.OneDay ? 'sm:w-72 md:w-96' : 'w-full min-w-72'
            }`}
         >
            {forecastType === ForecastType.OneDay ? (
               <>
                  <div className="flex flex-col">
                     {todaysWeather ? <WeatherDetails weather={todaysWeather} /> : Loading}
                  </div>
               </>
            ) : (
               <>
                  {weatherForecast && weatherForecast.length > 0
                     ? weatherForecast.map((weatherDetail, i) => (
                          <div key={i} className="flex flex-col">
                             <WeatherDetails weather={weatherDetail} />
                          </div>
                       ))
                     : Loading}
               </>
            )}
         </div>
      </div>
   );
};

export default ForecastDisplay;
