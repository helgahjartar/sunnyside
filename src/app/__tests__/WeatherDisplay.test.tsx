import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import WeatherDisplay from '../components/WeatherDisplay';

describe('WeatherDisplay', () => {
   test('matches snapshot when in loading state', () => {
      render(<WeatherDisplay />);

      expect(screen.getByTestId('weather-display')).toMatchSnapshot();
   });
});
