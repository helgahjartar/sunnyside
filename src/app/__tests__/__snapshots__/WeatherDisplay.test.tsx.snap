// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDisplay matches snapshot when in loading state 1`] = `
<div
  data-testid="weather-display"
>
  <div
    class="flex flex-row justify-between items-center"
  >
    <div
      class="h-12 text-black text-3xl font-bold w-full truncate py-2 px-4 capitalize"
    >
      Reykjav√≠k
      , 
      IS
    </div>
    <div
      class="flex align-middle"
    >
      <button
        class="min-h-10 bg-orange-300 hover:bg-orange-500 text-white text-lg font-bold py-2 px-4 rounded-lg shadow-lg transition duration-300 ease-in-out transform hover:scale-105"
      >
        See Current Weather
      </button>
    </div>
  </div>
  <div
    class="flex justify-between overflow-scroll mt-4 bg-[#F5ECDA] rounded-lg py-8 px-10 space-x-10 w-full min-w-72"
  >
    <div
      class="text-2xl font-semibold text-black"
    >
      Loading..
    </div>
  </div>
</div>
`;
